packages/vm/src/runTx.ts

```diff
@@ -205,20 +201,19 @@ async function _runTx(vm: VM, opts: RunTxOpts): Promise<RunTxResult> {
       throw Error(`Verkle access witness needed for execution of verkle blocks`)
     }
 
-    if (
-      !(vm.stateManager instanceof StatefulVerkleStateManager) &&
-      !(vm.stateManager instanceof StatelessVerkleStateManager)
-    ) {
+    if (!('verkleCrypto' in vm.stateManager)) {
```

Comment:
This should now also be made consistent, right? So this should check the `name` property (or if we rename that property to whatever it is renamed to) instead of checking for specific values in the state manager to derive which class type it is.